forge-game/src/main/java/forge/game/ability/effects/TokenEffectBase.java

```diff
@@ -191,6 +191,10 @@ protected TokenCreateTable makeTokenTable(TokenCreateTable tokenTable, final boo
                 if (sa.hasParam("RememberTokens")) {
                     host.addRemembered(moved);
                 }
+                // used for some reflexive trigger
+                if (sa.hasParam("RememberOriginalTokens") && moved.getRules().getNormalizedName().equals(sa.getParam("TokenScript"))) {
```

Comment:
@Hanmac
can we do a better check? though it does seem to work 🤔

Reply:
Compare with original TokenTable from before the Replacement Effect in line 85